### PR TITLE
docs: update link style and fix typos

### PR DESCRIPTION
**Pull request by shmpwk**
Signed-off-by: Shumpei Wakabayashi <shumpei.wakabayashi@tier4.jp>

## Description

Some documentaion links imported from https://autowarefoundation.gitlab.io/autoware.auto/AutowareAuto/ are dead.
I've fixed them and confirmed their links work properly.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [x] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Comment by shmpwk** [comment]
I understand temporary failing build-and-test-differential ci because of docker construction.

**Comment by kenji-miyake** [comment]
I guess it's due to https://github.com/autowarefoundation/autoware/issues/343.
I'm sorry but please wait for a while. :bow:

**Comment by shmpwk** [comment]
Some relative links dead by https://github.com/autowarefoundation/autoware.universe/pull/950/commits/7c477738845b78fc317009295968e06faf44127f.
[ [✖] ../../../trajectory_follower/design/trajectory_follower-mpc-design → Status: 400 ](https://github.com/autowarefoundation/autoware.universe/runs/6567020803?check_suite_focus=true#step:3:958)
[  [✖] ../../../trajectory_follower/design/trajectory_follower-design.md → Status: 400](https://github.com/autowarefoundation/autoware.universe/runs/6567020803?check_suite_focus=true#step:3:306)

So I specified absolute path for them by https://github.com/autowarefoundation/autoware.universe/pull/950/commits/5df2a5e3a1311ca7a469664ea5c60cb3eb72da65.

File: control/trajectory_follower_nodes/design/trajectory_follower-design.md [hunk 1]
```diff
@@ -8,12 +8,12 @@ Generate control commands to follow a given Trajectory.
...
-- @subpage lateral-controller-design : generates lateral control messages.
-- @subpage longitudinal-controller-design : generates longitudinal control messages.
-- @subpage latlon-muxer-design : combines the lateral and longitudinal control commands
+- [lateral-controller-design](lateral_controller-design.md) : generates lateral control messages.
+- [longitudinal-controller-design](longitudinal_controller-design.md) : generates longitudinal control messages.
+- [latlon-muxer-design](latlon_muxer-design.md) : combines the lateral and longitudinal control commands
   into a single control command.
 
-Core functionalities are implemented in the @subpage trajectory_follower-package-design package.
+Core functionalities are implemented in the [trajectory_follower](https://github.com/autowarefoundation/autoware.universe/blob/main/control/trajectory_follower/design/trajectory_follower-design.md) package.
```
Comment (kenji-miyake) [inline control/trajectory_follower_nodes/design/trajectory_follower-design.md:16]: If it's in the same repository, it's better to use relative links.

Related: https://github.com/autowarefoundation/autoware-documentation/pull/116#discussion_r876735966

File: control/trajectory_follower_nodes/design/lateral_controller-design.md [hunk 2]
```diff
@@ -23,7 +23,7 @@ The MPC uses a model of the vehicle to simulate the trajectory resulting from th
 The optimization of the control command is formulated as a Quadratic Program (QP).
 
 These functionalities are implemented in the `trajectory_follower` package
-(see @subpage trajectory_follower-mpc-design)
+(see [trajectory_follower-mpc-design](https://github.com/autowarefoundation/autoware.universe/blob/main/control/trajectory_follower/design/trajectory_follower-mpc-design.md))
```
Comment (kenji-miyake) [inline control/trajectory_follower_nodes/design/lateral_controller-design.md:26]: Same here.

**Comment by kenji-miyake** [comment]
Could you address some existing typos in this PR? :pray: 
image

If necessary, please register words to https://github.com/tier4/autoware-spell-check-dict/.

File: control/trajectory_follower_nodes/design/trajectory_follower-design.md (hunk 1 above)
Comment (shmpwk) [reply control/trajectory_follower_nodes/design/trajectory_follower-design.md]: I'm struggling with relative path...... 
All relative path are dead even if the docs is alive.
Could you give me some advise?

### Case 1:  [relative-path]/file-name
https://github.com/autowarefoundation/autoware.universe/commit/7c477738845b78fc317009295968e06faf44127f
[ [✖] ../../../trajectory_follower/design/trajectory_follower-mpc-design → Status: 400 ](https://github.com/autowarefoundation/autoware.universe/runs/6567020803?check_suite_focus=true#step:3:958)

when clicking trajectory_follower-mpc-design,

Screenshot from 2022-05-24 13-46-47

#### **docs is alive.**
Screenshot from 2022-05-24 13-47-07


### Case 2:  [relative-path]/file-name.md
https://github.com/autowarefoundation/autoware.universe/commit/7c477738845b78fc317009295968e06faf44127f
[  [✖] ../../../trajectory_follower/design/trajectory_follower-design.md → Status: 400](https://github.com/autowarefoundation/autoware.universe/runs/6567020803?check_suite_focus=true#step:3:306)

when clicking trajectory_follow-design,

Screenshot from 2022-05-24 13-47-21

docs is dead.
Screenshot from 2022-05-24 13-47-25


### Case 3:  [relative-path]/file-name.md/#heading
https://github.com/autowarefoundation/autoware.universe/pull/950/commits/313c1d818afb74bc32e485d8716af4b93cddcc59
[[✖] ../../../trajectory_follower/design/trajectory_follower-design.md/#trajectory-follower](https://github.com/autowarefoundation/autoware.universe/runs/6570597117?check_suite_focus=true#step:3:300)

docs is dead. 

### Case 4: [relative-path]/file-name/file-name.md#heading
https://github.com/autowarefoundation/autoware.universe/pull/950/commits/313c1d818afb74bc32e485d8716af4b93cddcc59
[[✖] ../../../trajectory_follower/design/trajectory_follower-mpc-design/trajectory_follower-mpc-design.md#mpc-trajectory-follower](https://github.com/autowarefoundation/autoware.universe/runs/6570597117?check_suite_focus=true#step:3:951)

docs is dead.

### Case 5: [relative-path]/file-name#heading
https://github.com/autowarefoundation/autoware.universe/pull/950/commits/a520ccfe39b2bd7287340e19baca5aa88ed04c89
[  [✖] ../../../trajectory_follower/design/trajectory_follower-mpc-design#mpc-trajectory-follower → Status: 400](https://github.com/autowarefoundation/autoware.universe/runs/6571088686?check_suite_focus=true#step:3:958)

#### **docs is alive!**

### Case6: [relative-path]/file-name.md#heading
https://github.com/autowarefoundation/autoware.universe/pull/950/commits/a520ccfe39b2bd7287340e19baca5aa88ed04c89
[[✖] ../../../trajectory_follower/design/trajectory_follower-design.md#trajectory-follower → Status: 400](https://github.com/autowarefoundation/autoware.universe/runs/6571088686?check_suite_focus=true#step:3:306)

docs is dead.

Comment (kenji-miyake) [reply control/trajectory_follower_nodes/design/trajectory_follower-design.md]: Okay, I'll try it!

Comment (kenji-miyake) [reply control/trajectory_follower_nodes/design/trajectory_follower-design.md]: Fixed in d0385a7c7f33bb59836ec65d7290828c41b47276.

The points are:
- The depth was wrong.
- You need `.md`.
- It's good to check with your IDE first, not with MkDocs.

Comment (shmpwk) [reply control/trajectory_follower_nodes/design/trajectory_follower-design.md]: Thank you!

> It's good to check with your IDE first,

I'll keep in mind.

File: perception/tensorrt_yolo/README.md [hunk 3]
```diff
@@ -8,6 +8,8 @@ This package detects 2D bounding boxes for target objects e.g., cars, trucks, bi
 
 ### Cite
 
+<!-- cspell:disable -->
```
Comment (kenji-miyake) [inline perception/tensorrt_yolo/README.md:11]: Can you use `cspell:ignore word1, word2` here instead?

Comment (shmpwk) [reply perception/tensorrt_yolo/README.md]: Ok !https://github.com/autowarefoundation/autoware.universe/pull/950/commits/6b7badefdd7e7e314540bfb984f089c7f2cfa93a

**Review by kenji-miyake — APPROVED** [review]
Great work! Thank you. :pray: 
I've updated the title a bit.

**Comment by shmpwk** [comment]
Thank you for your review!
Maybe unknown spell warnings should be fixed in another PR since it will be big change.
https://github.com/autowarefoundation/autoware.universe/runs/6573258797?check_suite_focus=true

**Comment by kenji-miyake** [comment]
@shmpwk Yes, it's fine. Since they are parameters, we have to care about the area of influence.

**Pull request merged**